eradiate/xarray/interp.py

```diff
@@ -0,0 +1,73 @@
+__all__ = ["film_to_angular"]
+
+import numpy as np
+import xarray as xr
+from numpy.typing import ArrayLike
+
+from ..frame import angles_to_direction
+from ..warp import uniform_hemisphere_to_square
+
+
+def film_to_angular(
+    da: xr.DataArray,
+    theta: ArrayLike,
+    phi: ArrayLike,
+    x_label: str = "x",
+    y_label: str = "y",
+    theta_label: str = "theta",
+    phi_label: str = "phi",
+) -> xr.DataArray:
+    """
+    Interpolate a hemispherical film data array on an angular grid.
+
+    Parameter ``da`` (:class:`xarray.DataArray`):
+        Data array with film coordinate pixels to interpolate on an angular grid.
+
+    Parameter ``theta`` (array-like):
+        List of zenith angle values.
+
+    Parameter ``phi`` (array-like):
+        List of azimuth angle values.
+
+    Parameter ``x_label`` (str):
+        Label for the width pixel coordinate.
+
+    Parameter ``y_label`` (str):
+        Label for the height pixel coordinate.
+
+    Parameter ``theta_label`` (str):
+        Label for the zenith angle coordinate.
+
+    Parameter ``phi_label`` (str):
+        Label for the azimuth angle coordinate.
+
+    Returns → :class:`xarray.DataArray`:
+        Data array interpolated on the specified angular grid.
+    """
+
+    # Interpolate values on angular grid
+    data = np.empty((len(phi), len(theta)))
+
+    # Map angular grid points to (x, y) space
+    for i, ph in enumerate(phi):
+        xs = np.empty_like(theta)
+        ys = np.empty_like(theta)
+
+        angles = np.array([[th, ph] for th in theta.ravel()])
+        directions = angles_to_direction(angles)
+        film_coords = uniform_hemisphere_to_square(directions)
+        xs.ravel()[:] = film_coords[:, 0]
```

Comment:
This looks strange to me. I would have expected `ravel` to return a new numpy object. But I suppose this actually creates a ravelled view to the original `xs` array?

Reply:
AFAIK `ravel()` returns a view; `flatten()` returns a copy (e.g. https://www.geeksforgeeks.org/differences-flatten-ravel-numpy/).